WordPress/openverse-frontend
# Replace `usage-data` store module with utils file

## Fixes


Fixes #1026 by @obulat (Alternative solution)

## Description


In this PR I propose to move actions of the `usage-data` out of Vuex and therefore delete this module, it does not have a state per se so we can save maintaining another module (in Vuex or Pinia). I created a utils file instead of a composable because it doesn't involve any state logic about the view.

This PR also removes the unused component `ReuseSurvey` and triggers the `sendDetailPageEvent` when clicking the source link on the **Image information** section.

## Testing Instructions

1. Check out this branch and enable Sentry temporarily in the [sentry-config.ts](https://github.com/WordPress/openverse-frontend/blob/usage-data-utils/src/utils/sentry-config.ts) file:
```diff
export const sentryConfig: ModuleConfiguration = {
  dsn:
    process.env.SENTRY_DSN || ...
-  disabled: !isProd,
+  disabled: false, 
  lazy: true,
}
```
2. Run the app with `ENABLE_INTERNAL_ANALYTICS=true pnpm dev`
3. Test the places from where events are sent:
- search results page: when searching for a term and/or clicking on a search result)
- image details page: click on author name or source, click on copy license info)
4. Observe if events are being sent

## Note
When we turn on analytics we should send these events from the audio page as well.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
